src/extensions/wcf/Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater/UpdaterFactory.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater
+{
+    public class UpdaterFactory
+    {
+        public static PackageUpdater? GetPackageUpdater(string path, ILogger logger)
+        {
+            try
+            {
+                XDocument doc = XDocument.Load(path);
+                PackageUpdater packageUpdater = new PackageUpdater(doc, logger);
+                return packageUpdater;
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, $"Error loading project file {path}");
+                return null;
+            }
+        }
+
+        public static ConfigUpdater? GetConfigUpdater(string path, ILogger logger)
+        {
+            try
+            {
+                XDocument doc = XDocument.Load(path);
+                ConfigUpdater configUpdater = new ConfigUpdater(doc, logger);
+                return configUpdater;
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, $"Error loading project file {path}");
+                return null;
+            }
+        }
+
+        public static SourceCodeUpdater? GetSourceCodeUpdater(string path, Dictionary<string, object> context, ILogger logger)
+        {
+            try
+            {
+                SyntaxTree tree = CSharpSyntaxTree.ParseText(File.ReadAllText(path));
+                SourceCodeUpdater sourceCodeUpdater = new SourceCodeUpdater(tree, UpdateTemplateCode(context, logger), logger);
+                return sourceCodeUpdater;
+            }
+            catch (Exception e)
```

Comment:
Do we expect any exceptions from the SourceCodeUpdater? I wouldn't expect that component to throw so we probably don't need the try/catch here.